Customizing/global/plugins/Services/Repository/RepositoryObject/ReportExamBio/classes/class.ilObjReportExamBio.php

```diff
@@ -0,0 +1,323 @@
+
+
+		$this->space = $table->prepareTableAndSetRelevantFields($this->space);
+		return $table;
+	}
+
+	public function filter() {
+		$pf = new \CaT\Filter\PredicateFactory();
+		$tf = new \CaT\Filter\TypeFactory();
+		$f = new \CaT\Filter\FilterFactory($pf, $tf);
+		$for_trainer_view = $this->forTrainerView();
+		$space = $this->space;
+		$sequence_args = array(
+			$f->dateperiod( $this->plugin->txt("dateperiod"), "")
+				->map(	function($start,$end) use ($f,$space) {
+							$pc = $f->dateperiod_timestamp_overlaps_predicate_fields
+								( $space->table('recent_pass_data')->field('testrun_finished_ts')
+								, $space->table('recent_pass_data')->field('testrun_finished_ts'));
+							return $pc($start,$end);
+						}
+						,$tf->cls("CaT\Filter\Predicates\Predicate")));
+
+		if($for_trainer_view) {
+			$sequence_args[] = $f->text($this->plugin->txt('lastname'))
+								->map(	function($lastname) use ($f,$space,$pf) {
+											if(!$lastname) {
+												return $pf->_TRUE();
+											}
+											$pc = $f->text_like_field($space->table('usr')->field('lastname'));
+											return $pc($lastname.'%');
+										}
+										,$tf->cls("CaT\Filter\Predicates\Predicate"));
+		}
+
+		$sequence_args[] = $f->multiselect($this->plugin->txt('test_title'),'',$this->getDistinctTests())
+							->map(	function($obj_ids) use ($f,$space,$pf) {
+										if(count($obj_ids) === 0) {
+											return $pf->_TRUE();
+										}
+										return $space->table('recent_pass_data')->field('obj_id')->IN($pf->list_int_by_array($this->convertMixedArrayToIntArray($obj_ids)));
+									}
+									,$tf->cls("CaT\Filter\Predicates\Predicate"));
+
+		$sequence_args[] = $f->multiselect($this->plugin->txt('pass_status'),'',$this->getDistinctPass())
+							->map(	function($pass) use ($f,$space,$pf) {
+										if(count($pass) === 0) {
+											return $pf->_TRUE();
+										}
+										return $space->table('recent_pass_data')->field('test_passed')->IN($pf->list_int_by_array($this->convertMixedArrayToIntArray($pass)));
+									}
+									,$tf->cls("CaT\Filter\Predicates\Predicate"));
+
+		return $f->sequence(call_user_func_array(array($f,'sequence'),$sequence_args)->map(
+				function (/*args*/) use ($for_trainer_view) {
+					$return = array();
+					$args = func_get_args();
+					if(count($args) > 0) {
+						reset($args);
+						$return['last_pass_datetime_predicate'] = current($args);
+						if($for_trainer_view) {
+							next($args);
+							$return['lastname_predicate'] = current($args);
+						}
+						next($args);
+						$return['test_title_predicate'] = current($args);
+						next($args);
+						$return['test_passed_predicate'] = current($args);
+					}
+					return $return;
+				}
+				,$tf->lst($tf->cls("CaT\Filter\Predicates\Predicate"))
+			));
+
+	}
+
+	public function initSpace() {
+		$aux = $this->tf->histUsertestrun('recent_pass_aux');
+		$aux = $aux->addConstraint($aux->field('hist_historic')->EQ()->int(0)
```

Comment:
This could be `something AND FALSE` if there are no relevant users, which is always False. Is this correct? Why not

```php
if (count($this->relevantUsers()) > 0) {
    $aux = $aux->addConstraint($aux->field('hist_historic')->EQ()->int(0));
}
else {
    $aux = $aux->addConstraint($this->pf->_FALSE());
}
```

This would be a lot more readable imo.